src/lib/translators/mongo.ts

```diff
@@ -227,11 +227,26 @@ export function _simplifyMongoPipeline(mongoSteps: Array<MongoStep>): Array<Mong
       stepOperator === '$project' || stepOperator === '$addFields' || stepOperator === '$match';
     if (isMergeable && lastStep[stepOperator] !== undefined) {
       for (const key in step[stepOperator]) {
-        // We do not want to merge two $project with common keys
+        /**
+         * In Mongo, exclusions cannot be combined with any inclusion, so if we
+         * have such an exclusion in a $project step, and that the previous one
+         * include any inclusion, we do not want to merge the steps.
+         */
+        if (stepOperator === '$project') {
+          const bool = step[stepOperator][key] === 0;
+          for (const lastKey in lastStep[stepOperator]) {
```

Comment:
Maybe this *for-loop* could be replaced by a call to `some`? I have something like this in mind:

```javascript
  const excluded = Number(step.$project[key] === 0);
  const merge = !Object.values(lastStep.$project).some(value => value === excluded);
```

Also, I have the feeling that this would even be enough:

```javascript
  const merge = !Object.values(lastStep.$project).some(value => value === step.$project[key]);
```

What do you think?

Reply:
Thank you for the `some` tip ! Much nicer and clearer.

Your second proposal does not test what we want.
An exclusion matches the pattern `column: 0` and an inclusion matches the pattern `column: \<something strictly different than 0\>`
We do not want two merge a step that includes an inclusion with another including at least one exclusion.

That being said, my tests are not enough => see you in the next commits ;)